app/src/main/java/io/zenandroid/onlinego/di/Modules.kt

```diff
@@ -176,6 +180,61 @@ private val viewModelsModule = module {
     )
   }
 
+  viewModel {
+    PuzzleDirectoryViewModel(
+      PuzzleRepository(get(), get()),
+      OGSRestService(get(), get(), get(), get()),
```

Comment:
you're specifically instantiating a new instance of `OGSRestService()` here, but I'm trying to keep that a singleton. you should be able to just `get()` here.